firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -535,4 +553,65 @@ private void redirectToPlayForAabUpdate(String downloadUrl)
     updateAppTaskCompletionSource.setResult(updateState);
     this.updateTask.updateProgress(updateState);
   }
+
+  private void showUpdateDialogIfNewReleaseAvailable() {
+    checkForUpdate()
+        .addOnSuccessListener(
+            latestRelease -> {
+              if (latestRelease != null) {
+                showUpdateAlertDialog(latestRelease);
+              } else {
+                updateToLatestReleaseTaskCompletionSource.setResult(null);
+              }
+            })
+        .addOnFailureListener(
+            e ->
+                setUpdateToLatestReleaseTaskCompletionError(
+                    new FirebaseAppDistributionException(
+                        Constants.ErrorMessages.NETWORK_ERROR, NETWORK_FAILURE)));
+  }
+
+  private void showUpdateAlertDialog(AppDistributionRelease latestRelease) {
+    Context context = firebaseApp.getApplicationContext();
+    AlertDialog alertDialog = new AlertDialog.Builder(currentActivity).create();
+    alertDialog.setTitle(context.getString(R.string.update_dialog_title));
+
+    alertDialog.setMessage(
+        String.format(
+            "Version %s (%s) is available.",
+            latestRelease.getDisplayVersion(), latestRelease.getBuildVersion()));
+
+    alertDialog.setButton(
+        AlertDialog.BUTTON_POSITIVE,
+        context.getString(R.string.update_yes_button),
+        (dialogInterface, i) -> {
+          try {
+            updateApp()
+                .addOnSuccessListener(
+                    updateState ->
+                        updateToLatestReleaseTaskCompletionSource.setResult(latestRelease))
+                .addOnFailureListener(
+                    e ->
+                        setUpdateToLatestReleaseTaskCompletionError(
```

Comment:
Same question here - if this error is a FirebaseAppDistributionException from updateApp, can we propagate that error instead of overwritting it?